### ISSUE TITLE
query_points = query_points.detach()

### ISSUE THREAD

**Issue by WAAutomation** [jbwang1997/OPUS#8]
Thank you for your outstanding work.
I wonder why you use detach in models\opus_transformer.py line 131
![image](https://github.com/user-attachments/assets/eea3dba1-3a17-42ff-b568-fc76edee6b80)

**Comment by jbwang1997** [comment]
Here, we want to truncate the regression gradient so that the regression loss of each layer only adjust this layer's regression offset.  The same operation is also applied in SparseBEV.

But, it not sure if detach will be beneficial to final accuracy. As far as I know, the model DINO aggregates two layer of regression loss gradients for offset adjustment.

**Comment by WAAutomation** [comment]
I found that SparseOCC detach both query_points and query_features. Is there any difference between these two approaches?
![image](https://github.com/user-attachments/assets/d62b80d5-a2ee-4f0e-b2ad-fb7449408ef2)

**Comment by jbwang1997** [comment]
By this way, each layer in SparseOcc is only optimzed according to this layer's losses. The processing in SparseOcc is certainly different with us. But, it's hard to tell which one is better before doing some ablation studies.

**Comment by WAAutomation** [comment]
thinks for your reply, I will do some ablations.
